File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        profit = 0
        buyAmount = prices[0]
        for i in range(1,len(prices)):
            if prices[i] < buyAmount:
                buyAmount = prices[i]
            else:
                profit = max(profit,prices[i]-buyAmount)
        return profit

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int profit = 0;
        int buyAmount = prices[0];
        for(int i =1;i<prices.length;i++){
            // if the stock is avaible for less on current day then buy it
            if(prices[i]<buyAmount){
                buyAmount = prices[i];
            }
            // sell the stock in other case
            else{
                profit = Math.max(profit,prices[i]-buyAmount);
            }
        }
        return profit;
    }
}